File: src/main/java/no/nav/fo/veilarbregistrering/oppgave/Virkedager.java
package no.nav.fo.veilarbregistrering.oppgave;

import java.time.LocalDate;

class Virkedager {

    private Virkedager() {
    }

    static LocalDate plussAntallArbeidsdager(LocalDate dato, int antallDager) {
        int teller = 0;
        int tellerDager = 0;

        while (teller != antallDager) {
            tellerDager++;
            if (Ukedag.erHelg(dato.plusDays(tellerDager))) {
                continue;
            }

            if (Helligdager.erHelligdag(dato.plusDays(tellerDager))) {
                continue;
            }

            teller++;
        }

        return dato.plusDays(tellerDager);
    }
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/oppgave/Virkedager.kt
package no.nav.fo.veilarbregistrering.oppgave

import no.nav.fo.veilarbregistrering.oppgave.Ukedag.erHelg
import no.nav.fo.veilarbregistrering.oppgave.Helligdager.erHelligdag
import no.nav.fo.veilarbregistrering.oppgave.Ukedag
import no.nav.fo.veilarbregistrering.oppgave.Helligdager
import java.time.LocalDate

internal object Virkedager {
    fun plussAntallArbeidsdager(dato: LocalDate, antallDager: Int): LocalDate {
        var teller = 0
        var tellerDager = 0
        while (teller < antallDager) {
            tellerDager++
            if (erHelg(dato.plusDays(tellerDager.toLong()))) {
                continue
            }
            if (erHelligdag(dato.plusDays(tellerDager.toLong()))) {
                continue
            }
            teller++
        }
        return dato.plusDays(tellerDager.toLong())
    }
}